app/jobs/course/material/download_job.rb

```diff
@@ -0,0 +1,30 @@
+class Course::Material::DownloadJob < ApplicationJob
+  include TrackableJob
+
+  protected
+
+  # Performs the download service.
+  #
+  # @param [Array<Course::Material|Course::Material::Folder>] contents The contents to be
+  #   downloaded.
+  # @param [String] filename The name of the zip file.
```

Comment:
OK, that makes sense. It should be the name of the folder containing all the files?